example/src/eyedrop/index.js

```diff
@@ -30,173 +31,182 @@ type Props = {
   onInit?: Function,
   onPickStart?: Function,
   onPickEnd?: Function,
-  passThrough?: string,
-  pickRadius?: { unit: 'pixel' | 'radius', amount: number }
+  colorsPassThrough?: string,
+  pickRadius?: number
 }
 
 type State = {
   colors: {
     rgb: string,
     hex: string
   },
-  pickingColorFromDocument: boolean
+  pickingColorFromDocument: boolean,
+  buttonDisabled: boolean
 }
 
-export default class EyeDropper extends React.Component<Props, State> {
-  constructor(props) {
-    super(props)
-    this.state = {
-      colors: { rgb: '', hex: '' },
-      pickingColorFromDocument: false
-    }
-    this.cursorActive = props.cursorActive ? props.cursorActive : 'copy'
-    this.cursorInactive = props.cursorInactive ? props.cursorInactive : 'auto'
-  }
+const EyeDropper = props => {
+
+  const [colors, setColors] = useState({ rgb: '', hex: '' });
+  const [pickingColorFromDocument, setPickingColorFromDocument] = useState(false);
+  const [buttonDisabled, setButtonDisabled] = useState(false);
+
+  const cursorActive = props.cursorActive ? props.cursorActive : 'copy'
+  const cursorInactive = props.cursorInactive ? props.cursorInactive : 'auto'
+
+  // initial stage of life cycle, catching errors
+  useEffect(() => {
+    const { onInit, pickRadius } = props
 
-  componentDidMount() {
-    const { onInit } = this.props
     if (onInit) { onInit() }
-  }
+    if (pickRadius) {
+      if (pickRadius < 0 || pickRadius > 450) {
+        throw new Error('pickRadius out of range: 0-450')
+      }
+    }
+  }, []);
 
-  componentWillReceiveProps(nextProps: Props) {
-    if(nextProps.once !== this.props.once && nextProps.once) {
-      document.removeEventListener('click', this.targetToCanvas)
-      document.body.style.cursor = this.cursorInactive
-      this.setState({ pickingColorFromDocument: false })
+  // setup listener for canvas picking click
+  useEffect(() => {
+    // setting "props.once" property
+    let once;
+    if (typeof props.once !== "undefined") { once = props.once; }
+    else { once = true; } // set default to true
```

Comment:
Maintain one style. :-D
```suggestion
    else { once = true } // set default to true
```